### ISSUE TITLE
Объеденить (виртуально) в одной вкладке несколько графиков одного инструмента с выбранными таймфреймами (и графику для них), чтобы не держать несколько вкладок одного инструмента с различными таймфреймами. 

### ISSUE THREAD

**Issue by Arseniy-Omsk** [SBProX/Test-Flight#60]
Конструктивное предложение: объеденить (виртуально) в одном окне несколько таймфреймов, чтобы не держать несколько вкладок одного инструмента с различными таймингами, а одно окно с заданными для загрузки несколькими рабочими таймингами. При чем рисунки, графические элементы, индикаторы создаются для одного окна и подстраиваются под каждый тайминг при переключении (по сути дублируются при отрисовку во все окна - т.е. имеем реально несколько окон, когда происходит отрисовка или настройка индикаторов то они пересчитываются под остальные тайминги, для индикаторов - скользящие к примеру, лучше делать пересчет (возможно галочку ставить - делать пересчет или нет), остальные и так должны работать).
Задавать необходимые для загрузки тайминги можно галочками напротив таймингов в настройках окна инструмента, переключение желательно в меню вверху тогда только загруженных таймингов или выпадающем как сейчас или горизонтальном (так как их не будет много в любом случае - каждый трейдер обычно контролирует привычное количество, а это обычно не более 4-х).
Зачем это нужно - 
1. Минимизация количества вкладок одного и того же инструмента.
2. Минимизация графической работы - прорисовки уровней, каналов и настройки каждого окна отдельно.

Я хотел сначала написать ответ на мое предыдущее предложение, но потом понял что то предложение проще закрыть так как оно сырое и действительно невыполнимое как небольшая доработка. Можно сказать оно неправильно сформулиированное. Если по существу - загрузка сейчас быстрая и можно сказать удовлетворительная.
Но количество вкладок графиков начинает зашкаливать. К примеру возьмем простой пример - надо отслеживать 6 инструментов и более и на каждый тайминг надо открывать новую вкладку - это получается на 6 инструментов 24 вкладки - это нормально? Я не про время загрузки, я про количество вкладок.
"Мы не храним информацию по ВСЕМ тайм-фреймам в оперативной памяти для быстрого переключения. Это крайне сложно, и не понятно зачем." 
Согласен.
Вот поэтому надо хранить информацию на каждый выбранный (руками - поставив галочку) отдельно таймфрейм.
Чтобы не плодить окна с графиками. Чтобы рисовать в каждом окне на все тайминги один уровень а не в каждом окне по 4 раза рисовать один и тот же уровень.
Так как сейчас приходится держать в SBpro по графику на каждый таймфрейм.
Вот количество графиков сразу увеличивается. Дневной - обязательно, час - тоже, 10 или 5 минут, и минутный.
А если надо несколько инструментов отслеживать?
Поэтому мое предложение это не глобальное изменение, а небольшое изменение интерфейса. Другая работа с окнами.
Возможно автокопирование графических элементов (автоудаление) в другие тайминги одного виртуального окна и инструмента.
Опять же, если к примеру сейчас построить канал на дневке, потом перейти на часовик и вернуться на дневку - канал слетает. Поэтому приходится не пользоваться преобразованиями, а рисовать всю графику каждый раз отдельно для каждого тайминга отдельно. Хотя тут можно было бы даже применить хитрость - возможность включать и отключать авторисование, т.е. дублирование рисунков в другие таймфреймы. 
Дневные уровни к примеру нужны на всех графиках, но небольшие уровни часовиков или минуток - не нужны на дневных графиках, так как таймфреймы это отдельные окна отображаемые виртуально на одной вкладке переключением таймфрейма, то возможность копировать во все окна каких то графических элементов разумно иметь возможность включать или выключать по желанию.

**Comment by staer-nk** [comment]
Согласен! Дельное предложение. Совмещение ТФ для ТС  было бы очень полезно.

**Comment by Abrikotin** [comment]
Поддерживаю это предложение.
